software/database/src/main/java/brooklyn/entity/database/mysql/MySqlClusterImpl.java

```diff
@@ -0,0 +1,445 @@
+                .entityFilter(Predicates.not(IS_MASTER))
+                .fromMembers()
+                .build());
+
+        addEnricher(Enrichers.builder()
+                .aggregating(MySqlNode.QUERIES_PER_SECOND_FROM_MYSQL)
+                .publishing(QUERIES_PER_SECOND_FROM_MYSQL_PER_NODE)
+                .fromMembers()
+                .computingAverage()
+                .defaultValueForUnreportedSensors(0d)
+                .build());
+    }
+
+    private void propagateMasterAttribute(AttributeSensor<?> att) {
+        addEnricher(Enrichers.builder()
+                .aggregating(att)
+                .publishing(att)
+                .computing(IfFunctions.ifPredicate(CollectionFunctionals.notEmpty())
+                        .apply(CollectionFunctionals.firstElement())
+                        .defaultValue(null))
+                .entityFilter(IS_MASTER)
+                .build());
+    }
+
+    @Override
+    protected EntitySpec<?> getFirstMemberSpec() {
+        final EntitySpec<?> firstMemberSpec = super.getFirstMemberSpec();
+        if (firstMemberSpec != null) {
+            return applyDefaults(firstMemberSpec, Suppliers.ofInstance(MASTER_SERVER_ID), MASTER_CONFIG_URL, false);
+        }
+
+        final EntitySpec<?> memberSpec = super.getMemberSpec();
+        if (memberSpec != null) {
+            if (!isKeyConfigured(memberSpec, MySqlNode.TEMPLATE_CONFIGURATION_URL.getConfigKey())) {
+                return EntitySpec.create(memberSpec)
+                        .configure(MySqlNode.MYSQL_SERVER_ID, MASTER_SERVER_ID)
+                        .configure(MySqlNode.TEMPLATE_CONFIGURATION_URL, MASTER_CONFIG_URL);
+            } else {
+                return memberSpec;
+            }
+        }
+
+        return EntitySpec.create(MySqlNode.class)
+                .displayName("MySql Master")
+                .configure(MySqlNode.MYSQL_SERVER_ID, MASTER_SERVER_ID)
+                .configure(MySqlNode.TEMPLATE_CONFIGURATION_URL, MASTER_CONFIG_URL);
+    }
+
+    @Override
+    protected EntitySpec<?> getMemberSpec() {
+        Supplier<Integer> serverIdSupplier = getAttribute(SLAVE_NEXT_SERVER_ID);
+
+        EntitySpec<?> spec = super.getMemberSpec();
+        if (spec != null) {
+            return applyDefaults(spec, serverIdSupplier, SLAVE_CONFIG_URL, true);
+        }
+
+        return EntitySpec.create(MySqlNode.class)
+                .displayName("MySql Slave")
+                .configure(MySqlNode.MYSQL_SERVER_ID, serverIdSupplier.get())
+                .configure(MySqlNode.TEMPLATE_CONFIGURATION_URL, SLAVE_CONFIG_URL)
+                // block inheritance, only master should execute the creation script
+                .configure(MySqlNode.CREATION_SCRIPT_URL, (String) null)
+                .configure(MySqlNode.CREATION_SCRIPT_CONTENTS, (String) null);
+    }
+
+    private EntitySpec<?> applyDefaults(EntitySpec<?> spec, Supplier<Integer> serverId, String configUrl, boolean resetCreationScript) {
```

Comment:
Might be nice to have `applyMasterDefaults` and `applySlaveDefaults` methods which call a third helper method with most of this content.

I like the clarity of that over just the `resetCreationScript` variable which is 1:1 with master / slave.